04_basic_struct/hello.c

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/printk.h>
+#include <linux/kobject.h>
+#include <linux/sysfs.h>
+#include <linux/list.h>
+#include <linux/slab.h> //kmalloc
+#include <linux/string.h>
+
+MODULE_AUTHOR("Oleh Yakymenko <oleh.o.yakymenko@globallogic.com");
+MODULE_DESCRIPTION("Linux kernel data structure");
+MODULE_LICENSE("Dual BSD/GPL");
+MODULE_VERSION("0.1");
+
+static LIST_HEAD(hello_list);
+struct hello_node {
+	struct list_head list;
+	char *data;
+};
+
+
+struct hello_node *list_create_node(const char *s, size_t count)
+{
+	struct hello_node *node = kmalloc(sizeof(*node),
+					GFP_KERNEL);
+
+	if (node == NULL)
+		pr_err("Element hasn`t created. Not enaugh memory for list entry\n");
+
+	node->data = kmalloc_array(count, sizeof(char), GFP_KERNEL);
+
+	strcpy(node->data, s);
+
+	return node;
+}
+
+void list_delete_node(struct hello_node *a_node)
+{
+	if (a_node == NULL)
+		return;
+
+	if (a_node->data != NULL)
+		kfree(a_node->data);
```

Comment:
If obj is NULL, no operation is performed. So the condition check is not required.